cmd/containerd-shim-runhcs-v1/service_internal.go

```diff
@@ -215,7 +224,14 @@ func (s *service) deleteInternal(ctx context.Context, req *task.DeleteRequest) (
 	if err != nil {
 		return nil, err
 	}
-	// TODO: We should be removing the task after this right?
+
+	if req.ExecID == "" {
+		// We've successfully deleted the init exec. Compute resources
+		// should be closed, layers umounted and resources released.
+		// This task should now be done.
+		s.taskOrPod.Store(getEmptyPodOrTask(s.isSandbox))
```

Comment:
If the service is tracking a pod (`s.isSandbox == true`), then there could be multiple tasks stored under `s.getPod().workloadTasks`, and emptying the `taskOrPod` store could prematurely close out other running tasks in the pod.
(I have a draft PR to address this: https://github.com/microsoft/hcsshim/pull/1234)

Reply:
Ahh! Then I'll remove this bit and simply clean up the execs. Your branch will take care of this properly.

Reply:
Done. Please have another look.